packages/devtools_app/lib/src/standalone_ui/ide_shared/property_editor/property_editor_view.dart

```diff
@@ -173,27 +184,290 @@ class _PropertyInput extends StatelessWidget {
-          controller: controller,
+          editProperty: editProperty,
         );
       case doubleType:
         return DoubleInput(
           property: property as NumericProperty,
-          controller: controller,
+          editProperty: editProperty,
         );
       case enumType:
         return EnumInput(
           property: property as FiniteValuesProperty,
-          controller: controller,
+          editProperty: editProperty,
         );
       case intType:
         return IntegerInput(
           property: property as NumericProperty,
-          controller: controller,
+          editProperty: editProperty,
         );
       case stringType:
-        return StringInput(property: property, controller: controller);
+        return StringInput(property: property, editProperty: editProperty);
       default:
         return Text(property.valueDisplay);
     }
   }
 }
+
+class _NoEditablePropertiesMessage extends StatelessWidget {
+  const _NoEditablePropertiesMessage({required this.name});
+
+  final String? name;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final fixedFontStyle = theme.fixedFontStyle.copyWith(
+      color: theme.colorScheme.primary,
+    );
+
+    return RichText(
+      text: TextSpan(
+        style: theme.regularTextStyle,
+        children: [
+          name == null
+              ? const TextSpan(text: 'The selected widget ')
+              : TextSpan(text: name, style: fixedFontStyle),
+          TextSpan(
+            text:
+                ' has no editable widget properties.\n\nThe Flutter Property Editor currently supports editing properties of type ',
+            style: theme.regularTextStyle,
+          ),
+          TextSpan(text: 'string', style: fixedFontStyle),
+          const TextSpan(text: ', '),
+          TextSpan(text: 'int', style: fixedFontStyle),
+          const TextSpan(text: ', '),
+          TextSpan(text: 'double', style: fixedFontStyle),
+          const TextSpan(text: ', '),
+          TextSpan(text: 'bool', style: fixedFontStyle),
+          const TextSpan(text: ', and '),
+          TextSpan(text: 'enum', style: fixedFontStyle),
+          const TextSpan(text: '.'),
+        ],
+      ),
+    );
+  }
+}
+
+class _WidgetNameAndDocumentation extends StatelessWidget {
+  const _WidgetNameAndDocumentation({required this.name, this.documentation});
+
+  final String name;
+  final String? documentation;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Row(
+          children: [
+            Padding(
+              padding: const EdgeInsets.only(bottom: denseSpacing),
+              child: Text(
+                name,
+                style: Theme.of(context).fixedFontStyle.copyWith(
+                  fontWeight: FontWeight.bold,
+                  fontSize: defaultFontSize + 1,
+                ),
+              ),
+            ),
+          ],
+        ),
+        Row(
```

Comment:
this should be in an expanded widget and the Column should have MainAxisSize.min

Reply:
Also, why do we need this row if there is only a single child?

Reply:
Added `MainAxisSize.min`, but without the row the text jumps to the left / right when the doc is collapsed / expanded